src/System.Private.CoreLib/shared/System/Diagnostics/DebugProvider.cs

```diff
@@ -73,19 +73,31 @@ public virtual void WriteLine(string? message)
         private sealed class DebugAssertException : Exception
         {
             internal DebugAssertException(string? stackTrace) :
-                base(Environment.NewLine + stackTrace)
+                base(stackTrace)
```

Comment:
You can delete the constructors with 1 and 2 arguments, and just change the calling code to call the overload with 3 argument. The calling code has logic that is redundant with the `Terminate`  that is here now.